message/result_metadata.go

```diff
@@ -380,3 +380,63 @@ func haveSameTable(cols []*ColumnMetadata) bool {
 	}
 	return true
 }
+
+func cloneColumnMetadata(cm *ColumnMetadata) *ColumnMetadata {
```

Comment:
How about we turn these functions into methods? (They can stay private)

Reply:
done, since they are now "standard" `Clone()` functions I decided to make them public for consistency purposes